src/main/java/dbaccessor/chat/ChatAccessor.java

```diff
@@ -77,20 +78,59 @@ public Chat getChat(long chatId) {
 
     /**
-     * Returns details of all Chats which the given User id engaged in.
+     * Returns details of all Chats which the given UserId engaged in.
      * Details include:
      * <ol>
      * <li> ChatId </li>
+     * <li> Creation Timestamp of Chat </li>
      * <li> LastSentMessageId </li>
-     * <li> Creation Timestamp </li>
+     * <li> Creation Timestamp of LastSentMessage </li>
+     * <li> Username of Second User in Chat </li>
+     * <li> MobileNo of Second User in Chat </li>
      * </ol>
      */
-    public List<Chat> getChatsForUser(User user) {
+    public List<ListChats.AllInfoForListChats> getAllInfoForListChats(long userId) {
+        String sqlStatment = "SELECT "
+                                + "WithUserID.ChatID AS ChatID, " 
+                                + "WithUserID.ChatCreationTS AS ChatCreationTS, " 
+                                + "WithUserID.LastSentMessageID AS LastSentMessageID, " 
+                                + "WithUserID.LastSentMessageCreationTS AS LastSentMessageCreationTS, " 
+                                + "User.Username AS Username, " 
+                                + "User.MobileNo AS MobileNo " 
+                            + "FROM (" 
+                                + "SELECT " 
+                                    + "WithLastSentMessageCreationTS.ChatID AS ChatID, " 
+                                    + "WithLastSentMessageCreationTS.ChatCreationTS AS ChatCreationTS, " 
+                                    + "WithLastSentMessageCreationTS.LastSentMessageID AS LastSentMessageID, " 
+                                    + "WithLastSentMessageCreationTS.LastSentMessageCreationTS AS LastSentMessageCreationTS, " 
+                                    + "UserChat.UserID AS UserID " 
+                                + "FROM (" 
+                                    + "SELECT " 
+                                        + "ChatsOfUser.ChatID AS ChatID, " 
+                                        + "ChatsOfUser.ChatCreationTS AS ChatCreationTS, " 
+                                        + "ChatsOfUser.LastSentMessageID AS LastSentMessageID, " 
+                                        + "Message.CreationTS AS LastSentMessageCreationTS " 
+                                    + "FROM (" 
+                                        + "SELECT " 
+                                        + "UserChat.ChatID AS ChatID, " 
+                                        + "Chat.CreationTS AS ChatCreationTS, " 
+                                        + "Chat.LastSentMessageID AS LastSentMessageID " 
+                                        + "FROM UserChat " 
+                                        + "INNER JOIN Chat " 
+                                        + "ON UserChat.ChatID = Chat.ChatID " 
+                                        + "WHERE UserChat.UserID = @userId" 
+                                        + ") AS ChatsOfUser " 
+                                    + "LEFT OUTER JOIN Message " 
```

Comment:
Instead of joining 2 tables (`UserChat` and `Chat`) and then joining its result with `Message`, can we not join the three tables in a single query. I think that should help in some early filtering.

Reply:
Done